kernel/interrupt_controller/src/lib.rs

```diff
@@ -0,0 +1,388 @@
+        #[cfg(target_arch = "x86_64")] {
+            // No priority support on x86_64
+            Priority
+        }
+    }
+
+    pub fn set_local_interrupt_priority(&self, num: LocalInterruptNumber, priority: Priority) {
+        get_int_ctlr!(int_ctlr, set_local_interrupt_priority);
+
+        #[cfg(target_arch = "aarch64")] {
+            int_ctlr.set_interrupt_priority(num.0, priority);
+        }
+
+        #[cfg(target_arch = "x86_64")] {
+            // No priority support on x86_64
+            let _ = priority;
+        }
+    }
+
+    pub fn is_local_interrupt_enabled(&self, num: LocalInterruptNumber) -> bool {
+        get_int_ctlr!(int_ctlr, is_local_interrupt_enabled);
+
+        #[cfg(target_arch = "aarch64")] {
+            int_ctlr.get_interrupt_state(num.0)
+        }
+
+        #[cfg(target_arch = "x86_64")] {
+            todo!()
+        }
+    }
+
+    pub fn enable_local_interrupt(&self, num: LocalInterruptNumber, enabled: bool) {
+        get_int_ctlr!(int_ctlr, enable_local_interrupt);
+
+        #[cfg(target_arch = "aarch64")] {
+            int_ctlr.set_interrupt_state(num.0, enabled);
+        }
+
+        #[cfg(target_arch = "x86_64")] {
+            todo!()
+        }
+    }
+
+    /// Sends an inter-processor interrupt to a specific CPU.
+    pub fn send_ipi(&self, destination: InterruptDestination) {
+        get_int_ctlr!(int_ctlr, send_ipi);
+
+        #[cfg(target_arch = "aarch64")] {
+            int_ctlr.send_ipi(destination.local_number.0, IpiTargetCpu::Specific(destination.cpu));
+        }
+
+        #[cfg(target_arch = "x86_64")] {
+            int_ctlr.send_ipi(destination.local_number.0, LapicIpiDestination::One(destination.cpu.into()))
+        }
+    }
+
+    /// Reads the minimum priority for an interrupt to reach this CPU.
+    ///
+    /// Note: aarch64-only, at the moment.
+    pub fn get_minimum_priority(&self) -> Priority {
+        get_int_ctlr!(int_ctlr, get_minimum_priority);
+
+        #[cfg(target_arch = "aarch64")] {
+            int_ctlr.get_minimum_priority()
+        }
+
+        #[cfg(target_arch = "x86_64")] {
+            // No priority support on x86_64
+            Priority
+        }
+    }
+
+    /// Changes the minimum priority for an interrupt to reach this CPU.
+    ///
+    /// Note: aarch64-only, at the moment.
+    pub fn set_minimum_priority(&self, priority: Priority) {
+        get_int_ctlr!(int_ctlr, set_minimum_priority);
+
+        #[cfg(target_arch = "aarch64")] {
+            int_ctlr.set_minimum_priority(priority)
+        }
+
+        #[cfg(target_arch = "x86_64")] {
+            // No priority support on x86_64
+            let _ = priority;
+        }
+    }
+
+    /// Aarch64-specific way to read the current pending interrupt number & priority.
+    #[cfg(target_arch = "aarch64")]
+    pub fn acknowledge_interrupt(&self) -> (LocalInterruptNumber, Priority) {
+        get_int_ctlr!(int_ctlr, acknowledge_interrupt);
+
+        #[cfg(target_arch = "aarch64")] {
```

Comment:
since the whole function is aarch64-only, you can remove this redundant cfg line.